cli/replicasets/join.go

```diff
@@ -0,0 +1,106 @@
+package replicasets
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/tarantool/cartridge-cli/cli/common"
+	"github.com/tarantool/cartridge-cli/cli/context"
+)
+
+func Join(ctx *context.Ctx, args []string) error {
+	var err error
+
+	if ctx.Replicasets.ReplicasetName == "" {
+		return fmt.Errorf("Please, specify replicaset name via --replicaset flag")
+	}
+
+	if err := FillCtx(ctx); err != nil {
+		return err
+	}
+
+	if ctx.Replicasets.JoinInstancesNames, err = common.GetInstancesFromArgs(args, ctx); err != nil {
+		return err
+	}
+
+	if len(ctx.Replicasets.JoinInstancesNames) == 0 {
+		return fmt.Errorf("Please, specify at least one instance name")
+	}
+
+	log.Infof(
+		"Join instance(s) %s to replicaset %s",
+		strings.Join(ctx.Replicasets.JoinInstancesNames, ", "),
+		ctx.Replicasets.ReplicasetName,
+	)
+
+	instancesConf, err := getInstancesConf(ctx)
+	if err != nil {
+		return fmt.Errorf("Failed to get instances configuration: %s", err)
+	}
+
+	conn, err := getControlConn(instancesConf, ctx, ctx.Replicasets.JoinInstancesNames)
+	if err != nil {
+		return fmt.Errorf("Failed to connect to Tarantool instance: %s", err)
+	}
+
+	topologyReplicasets, err := getTopologyReplicasets(conn)
+	if err != nil {
+		return fmt.Errorf("Failed to get current topology replicasets: %s", err)
+	}
+
+	editReplicasetOpts, err := getJoinInstancesEditReplicasetsOpts(
+		ctx.Replicasets.ReplicasetName, ctx.Replicasets.JoinInstancesNames,
+		topologyReplicasets, instancesConf,
+	)
+	if err != nil {
+		return fmt.Errorf("Failed to get edit_topology options for joining instances: %s", err)
+	}
+
+	if _, err = editReplicaset(conn, editReplicasetOpts); err != nil {
+		return fmt.Errorf("Failed to join instances: %s", err)
+	}
+
+	log.Infof(
+		"Instance(s) %s successfully joined to replicaset %s",
+		strings.Join(ctx.Replicasets.JoinInstancesNames, ", "),
+		ctx.Replicasets.ReplicasetName,
+	)
+
+	return nil
+}
+
+func getJoinInstancesEditReplicasetsOpts(replicasetAlias string, joinInstancesNames []string,
+	topologyReplicasets *TopologyReplicasets, instancesConf *InstancesConf) (*EditReplicasetOpts, error) {
+
+	editReplicasetOpts := EditReplicasetOpts{}
+
+	topologyReplicaset := topologyReplicasets.GetByAlias(replicasetAlias)
+	if topologyReplicaset != nil {
+		editReplicasetOpts.ReplicasetUUID = topologyReplicaset.UUID
+	} else {
+		editReplicasetOpts.ReplicasetAlias = replicasetAlias
+	}
```

Comment:
Why can't you return error here. This case looks strange.

Reply:
This is very important code (I will add a comment):

* if replicaset already exists, we have to specify it's URI to join new instances to it
* if replicaset doesn't exist, we specify it's alias to create a new replicaset with specified instances

Reply:
Yes, please, do